### ISSUE TITLE
Benefit of this plugin vs rimraf

### ISSUE THREAD

**Issue by garygreen** [johnagan/clean-webpack-plugin#75]
Just trying to understand what benefit you get from using this plugin over just a simple rimraf command to clean your dist?

I would have thought that the plugin would be "smart" and only clean stuff that Webpack no longer needs / isn't detected in the build chain any longer... some kind of prune type functionality, but my understanding is this plugin offers no much more functionality other than a basic remove (with some options to exclude etc)?

**Comment by chrisblossom** [comment]
I came to the exact same conclusion. See discussion https://github.com/johnagan/clean-webpack-plugin/issues/60.

I created a plugin that keeps track of, and only removes unused webpack assets (no exclude required): [chrisblossom/clean-self-webpack-plugin](https://github.com/chrisblossom/clean-self-webpack-plugin). I suggested merging it with this plugin, which I am still open to/think the community would benefit from.

**Comment by garygreen** [comment]
@chrisblossom well that's more like it! Nice easy to understand code that does something useful with the webpack build chain.

That's really why I'm just struggling to understand why this plugin has become popular - not to stamp on the hard work done on it, but if you take a look at the code it's really not doing anything special.

What's spured me on to take a deep looker at plugins is because I've been moving from Gulp to Webpack and it's pretty common to have this kind of thing to determine what to run in dev/production:

```
production ? <run this task | plugin> : noop
```

So I was wondering if there was a `NoopPlugin` for webpack of some kind - and woah, there is! https://github.com/QingWei-Li/noop-webpack-plugin

Intrigued, I took a look at the code:

```php
module.exports = function () {
  return {
    apply: function () {}
  }
}
```

That's it. Literally does nothing and someone has created a plugin for it! So I just copied and pasted it at the top of my config and renamed it `noop`.

I get similar kind of vibes from this plugin, at the meat of it it's just doing `rimraf` - presumably over a `dist` directory. The only other stuff it's doing is options for including/excluding certain directories - but  my thoughts on that is the `dist` directory or whatever folder your adding really should be totally destructible, you shouldn't be storing any files or stuff in dist that should be ignored to begin with.

I dunno. I probably just have a limited understanding of it's greater usage. 🐧
